metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/AbstractConnectionProvider.java

```diff
@@ -0,0 +1,90 @@
+package eu.europeana.enrichment.rest.client;
+
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+
+public abstract class AbstractConnectionProvider {
+
+  /**
+   * The default value of the maximum amount of time, in milliseconds, we wait for a connection to a
+   * resource before timing out. It's currently set to {@value AbstractConnectionProvider#DEFAULT_CONNECT_TIMEOUT}
+   * milliseconds.
+   */
+  public static final int DEFAULT_CONNECT_TIMEOUT = 10_000;
+
+  /**
+   * The default value of the maximum amount of time, in milliseconds, we wait for the response
+   * before timing out. It's currently set to {@value AbstractConnectionProvider#DEFAULT_RESPONSE_TIMEOUT}
+   * milliseconds.
+   */
+  public static final int DEFAULT_RESPONSE_TIMEOUT = 60_000;
+
+  /**
+   * The default value of the batch size with which we query the enrichment service. It's currently
+   * set to {@value AbstractConnectionProvider#DEFAULT_BATCH_SIZE_ENRICHMENT} values.
+   */
+  public static final int DEFAULT_BATCH_SIZE_ENRICHMENT = 20;
+
+  private int connectTimeout = DEFAULT_CONNECT_TIMEOUT;
+  private int responseTimeout = DEFAULT_RESPONSE_TIMEOUT;
+  protected int batchSizeEnrichment = DEFAULT_BATCH_SIZE_ENRICHMENT;
+
+  /**
+   * Set the maximum amount of time, in milliseconds, we wait for a connection before timing out.
+   * The default (when not calling this method) is {@value AbstractConnectionProvider#DEFAULT_CONNECT_TIMEOUT}
+   * milliseconds.
+   *
+   * @param connectTimeout The maximum amount of time, in milliseconds, we wait for a connection
+   * before timing out. If not positive, this signifies that the connection does not time out.
+   * @return This instance, for convenience.
+   */
+  public AbstractConnectionProvider setConnectTimeout(int connectTimeout) {
+    this.connectTimeout = connectTimeout;
+    return this;
```

Comment:
I guess since it's not a "Builder" anymore those methods don't need to return "this".
So they should either be normal setters or part of the constructor or both.
Same goes for the implemented classes. DereferencerProvider, EnricherProvider

Reply:
You are right, I will fix this